### PR TITLE
Another round of misc cleanup

### PR DESCRIPTION
**Pull request by hlinnaka**
Misc stuff that have caught my eye while working on other stuff. See commit messages for details.

### CODE REVIEW

File: src/backend/storage/lmgr/proc.c [hunk 1]
```diff
@@ -1059,22 +1061,6 @@ AuxiliaryProcKill(int code, Datum arg)
 	update_spins_per_delay();
 
 	SpinLockRelease(ProcStructLock);
-
-	/*
-	 * If the parent process of this auxiliary process does not exist, we
-	 * have trouble. Besides the obvious case that the postmaster is gone,
-	 * this could happen to filerep subprocesses when the filerep main
```
Comment (BaiShaoqi) [inline src/backend/storage/lmgr/proc.c]: What is a filerep process? Filerep is used for?

**Comment by hlinnaka** [comment]
On 27/12/2018 19:10, BaiShaoqi wrote:
> What is a filerep process? Filerep is used for?
"Filerep", or "file replication", was a system for replicating tables 
from master nodes to mirrors, in GPDB 5 and below. It was completely 
ripped out in GDPB 6, and replaced with using PostgreSQL's WAL replication.

A "filerep process" was an auxiliary backend process, that handled the 
file replication. There were actually several different helper processes 
involved, "filerep process" was a generic term that covered all of them.

- Heikki

File: src/include/cdb/cdbexplain.h [hunk 2]
```diff
@@ -16,16 +16,20 @@
...
 struct CdbDispatchResults;              /* #include "cdb/cdbdispatchresult.h" */
-struct EState;                          /* #include "nodes/execnodes.h" */
 struct PlanState;                       /* #include "nodes/execnodes.h" */
 struct QueryDesc;                       /* #include "executor/execdesc.h" */
-struct StringInfoData;                  /* #include "lib/stringinfo.h" */
 
 struct CdbExplain_ShowStatCtx;          /* private, in "cdb/cdbexplain.c" */
-struct PlannedStmt;
+
+typedef struct
```
Comment (ashwinstar) [inline src/include/cdb/cdbexplain.h]: formatting seems funky atleast on GitHub.

Comment (hlinnaka) [reply src/include/cdb/cdbexplain.h]: It's just github, if you're referring to the misaligned field names. Have a look at the "View file" view, it shows the tabs with correct width: https://github.com/greenplum-db/gpdb/blob/39772f818f3d6efaa0e22512b93efe260f6b3cba/src/include/cdb/cdbexplain.h

Comment (ashwinstar) [reply src/include/cdb/cdbexplain.h]: Thanks

**Pull request merged**